demisto_sdk/commands/openapi_codegen/openapi_codegen.py

```diff
@@ -0,0 +1,837 @@
+            command_description = command['description']
+            commands.append(XSOARIntegration.Script.Command(command_name, command_description, args, outputs))
+
+        script = ''
+        if not no_code:
+            script = self.get_python_code()
+
+        commonfields = XSOARIntegration.CommonFields(self.configuration['name'], -1)
+        name = self.configuration['name']
+        display = self.configuration['name']
+        category = self.configuration['category']
+        description = self.configuration['description']
+        url = self.configuration['url']
+        configurations = [XSOARIntegration.Configuration(display=f'Server URL (e.g. {url})',
+                                                         name='url',
+                                                         defaultvalue=url,
+                                                         type_=0,
+                                                         required=True)]
+        if not isinstance(self.configuration['auth'], list):
+            self.configuration['auth'] = [self.configuration['auth']]
+
+        if API_KEY_AUTH_TYPE in self.configuration['auth']:
+            configurations.append(XSOARIntegration.Configuration(display='API Key',
+                                                                 name='api_key',
+                                                                 required=True,
+                                                                 type_=4))
+        if CREDENTIALS_AUTH_TYPE in self.configuration['auth']:
+            configurations.append(XSOARIntegration.Configuration(display='Username',
+                                                                 name='credentials',
+                                                                 required=True,
+                                                                 type_=9))
+        if self.configuration.get('fetch_incidents', False):
+            configurations.extend([
+                XSOARIntegration.Configuration(display='Fetch incidents',
+                                               name='isFetch',
+                                               type_=8,
+                                               required=False),
+                XSOARIntegration.Configuration(display='Incident type',
+                                               name='incidentType',
+                                               type_=13,
+                                               required=False),
+                XSOARIntegration.Configuration(display='Maximum number of incidents per fetch',
+                                               name='max_fetch',
+                                               defaultvalue='10',
+                                               type_=0,
+                                               required=False),
+                XSOARIntegration.Configuration(display='API Key',

```

Comment:
what

Reply:
what?

Reply:
why is this here? dont we add credentials / api key based on auth type?

Reply:
oops